tests/filesystem.go

```diff
@@ -0,0 +1,517 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package blackbox
+
+import (
+	"bufio"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/coreos/ignition/tests/types"
+)
+
+func mustRun(t *testing.T, command string, args ...string) []byte {
+	out, err := exec.Command(command, args...).CombinedOutput()
+	if err != nil {
+		t.Fatal(command, err, string(out))
+	}
+	return out
+}
+
+func prepareRootPartitionForPasswd(t *testing.T, partitions []*types.Partition) {
+	for _, p := range partitions {
+		if p.Label == "ROOT" {
+			dirs := []string{
+				filepath.Join(p.MountPath, "home"),
+				filepath.Join(p.MountPath, "usr", "bin"),
+				filepath.Join(p.MountPath, "usr", "sbin"),
+				filepath.Join(p.MountPath, "usr", "lib64"),
+				filepath.Join(p.MountPath, "etc"),
+			}
+			for _, dir := range dirs {
+				_ = os.MkdirAll(dir, 0755)
+			}
+
+			symlinks := []string{"lib64", "bin", "sbin"}
+			for _, symlink := range symlinks {
+				_ = os.Symlink(
+					filepath.Join(p.MountPath, "usr", symlink),
+					filepath.Join(p.MountPath, symlink))
+			}
+
+			copies := [][2]string{
+				{
+					"/etc/ld.so.cache", filepath.Join(p.MountPath, "etc"),
+				},
+				{
+					"/lib64/libblkid.so.1", filepath.Join(p.MountPath, "usr", "lib64"),
+				},
+				{
+					"/lib64/libpthread.so.0", filepath.Join(p.MountPath, "usr", "lib64"),
+				},
+				{
+					"/lib64/libc.so.6", filepath.Join(p.MountPath, "usr", "lib64"),
+				},
+				{
+					"/lib64/libuuid.so.1", filepath.Join(p.MountPath, "usr", "lib64"),
+				},
+				{
+					"/lib64/ld-linux-x86-64.so.2", filepath.Join(p.MountPath, "usr", "lib64"),
+				},
+				{
+					"/lib64/libnss_files.so.2", filepath.Join(p.MountPath, "usr", "lib64"),
+				},
+				{
+					"bin/amd64/id", filepath.Join(p.MountPath, "usr", "bin"),
+				},
+				{
+					"bin/amd64/useradd", filepath.Join(p.MountPath, "usr", "sbin"),
+				},
+				{
+					"bin/amd64/usermod", filepath.Join(p.MountPath, "usr", "sbin"),
+				},
+			}
+
+			for _, cp := range copies {
+				_, _ = exec.Command("cp", cp[0], cp[1]).CombinedOutput()
+			}
+		}
+	}
+}
+
+func getRootLocation(partitions []*types.Partition) string {
+	for _, p := range partitions {
+		if p.Label == "ROOT" {
+			return p.MountPath
+		}
+	}
+	return ""
+}
+
+func removeFile(t *testing.T, imageFile string) {
+	err := os.Remove(imageFile)
+	if err != nil {
+		t.Log(err)
+	}
+}
+
+func removeMountFolders(t *testing.T, partitions []*types.Partition) {
+	for _, p := range partitions {
+		err := os.RemoveAll(p.MountPath)
```

Comment:
If I understand correctly, the mountpoints should be empty, so it'd be safer to just make this a `Remove`.